asdf_astropy/extension.py

```diff
@@ -0,0 +1,50 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+import os
+
+import asdf
+
+# Make sure that all tag implementations are imported by the time we create
+# the extension class so that _astropy_asdf_types is populated correctly. We
+# could do this using __init__ files, except it causes pytest import errors in
+# the case that asdf is not installed.
+'''
+from .tags.coordinates.angle import *  # noqa
+from .tags.coordinates.frames import *  # noqa
+from .tags.coordinates.earthlocation import *  # noqa
+from .tags.coordinates.skycoord import *  # noqa
+from .tags.coordinates.representation import *  # noqa
+from .tags.coordinates.spectralcoord import *  # noqa
+from .tags.fits.fits import *  # noqa
+from .tags.table.table import *  # noqa
+from .tags.time.time import *  # noqa
+from .tags.time.timedelta import *  # noqa
+'''
+from .tags.transform.basic import *  # noqa
+from .tags.transform.polynomial import *  # noqa
+'''
+from .tags.transform.compound import *  # noqa
+from .tags.transform.functional_models import *  # noqa
+from .tags.transform.physical_models import *  # noqa
+#from .tags.transform.math import *  # noqa
+from .tags.transform.powerlaws import *  # noqa
+#from .tags.transform.projections import *  # noqa
+from .tags.transform.tabular import *  # noqa
+from .tags.unit.quantity import *  # noqa
+from .tags.unit.unit import *  # noqa
+from .tags.unit.equivalency import *  # noqa
+'''
+
+_asdf_format_types = {ShiftType, ScaleType, Linear1DType, MultiplyType,
+                      OrthoPolynomialType, PolynomialTypeBase, OrthoPolynomialType}
+
+
+__all__ = ['TransformConverterProvider']
+
+
+class TransformConverterProvider(asdf.AsdfConverterProvider):
+    converter_classes = list(_asdf_format_types)
+
+
+asdf.configure(converter_providers=[TransformConverterProvider()])
```

Comment:
I don't think we want to call `configure` from within a library, since that will override other providers that the user might have installed.  The entry point should take care of this for us (but maybe it's broken right now, I can't remember if I tested that specifically).

Reply:
A quick test shows it doesn't work without it.

Reply:
Ah okay thanks, I'll investigate

Reply:
I think I found the issue, left a comment here:

https://github.com/astropy/asdf-astropy/pull/2/files#r441786043